tests/utils/stl/test/tests.py

```diff
@@ -7,215 +7,199 @@
-        self.env_num = env_num
-        self.skipped = False
-        Test.__init__(self, suite, path_in_suite, test_config, file_path)
-
-        self._configure_expected_result(suite, path_in_suite, lit_config,
-                                        test_config, env_num)
-        if self.skipped:
+    def __init__(self, suite, pathInSuite, litConfig, testConfig,
+                 envlstEntry, envNum):
+        self.compileFlags = []
+        self.cxx = None
+        self.envNum = envNum
+        self.fileDependencies = []
+        self.flags = []
+        self.isenseRspPath = None
+        self.linkFlags = []
+        self.testType = None
+        Test.__init__(self, suite, pathInSuite, copy.deepcopy(testConfig), None)
+
+        self._configureExpectedResult(suite, litConfig)
+        if self.result:
             return
 
-        self._configure_cxx(lit_config, envlst_entry, default_cxx)
-
-        use_edg = False
-        for flag in chain(self.cxx.flags, self.cxx.compile_flags):
-            if flag[1:] == 'clr:pure':
-                self.requires.append('clr_pure') # TRANSITION, GH-798
-            elif flag[1:] == 'clr':
-                self.requires.append('clr') # TRANSITION, GH-797
-            elif flag[1:] == 'BE':
-                self.requires.append('edg') # available for x86, see config.py
-                use_edg = True
-
-        if not use_edg and self.cxx.edg_drop is not None:
-            self.skipped = True
-
-    def getOutputDir(self):
-        return Path(os.path.join(
-            self.suite.getExecPath(self.path_in_suite[:-1]))) / self.env_num
+        self._handleEnvlst(litConfig, envlstEntry)
+        self._configureTestType()
+        _, tmpBase = self.getTempPaths()
 
-    def getOutputBaseName(self):
-        return self.path_in_suite[-2]
+    def _configureTestType(self):
+        self.testType = TestType.UNKNOWN
+        filename = self.path_in_suite[-1]
 
-    def getExecDir(self):
-        return self.getOutputDir()
+        if filename.endswith('.fail.cpp'):
+            self.testType = self.testType | TestType.FAIL
+        elif filename.endswith('.pass.cpp'):
+            self.testType = self.testType | TestType.PASS
+        else:
+            self.testType = self.testType | TestType.PASS
+
+        shortenedFlags = [flag[1:] for flag in chain(self.flags, self.compileFlags, self.linkFlags)]
+        if 'analyze:only' in shortenedFlags or 'c' in shortenedFlags or \
+                filename.endswith(('.compile.pass.cpp', 'compile.fail.cpp')):
+            self.testType = self.testType | TestType.COMPILE
+        elif filename.endswith(('.link.pass.cpp', '.link.fail.cpp')):
+            self.testType = self.testType | TestType.LINK
+        elif filename.endswith(('run.fail.cpp','run.pass.cpp')):
+            self.testType = self.testType | TestType.RUN
+        elif filename.endswith('.fail.cpp'):
+            self.testType = self.testType | TestType.COMPILE

```

Comment:
Wait, so `.fail.cpp` means the same thing as `.compile.fail.cpp`, instead of having the `RUN` semantic like `.pass.cpp`? That seems like a nasty inconsistency that will silently create an absence of test coverage when misused, which is likely to happen given that `.fail.cpp` tests in libc++ are run.

Reply:
`.fail.cpp` tests are meant to be the same as `compile.fail.cpp` tests and will eventually all be replaced.

Reply:
https://github.com/llvm/llvm-project/blob/5b30d9adc0536eee7fe0f164a550084916899acc/libcxx/utils/libcxx/test/format.py#L138